warning: 'prog_char' is deprecated

using Arduino 1.6.7 on OS X 10.11.3

Getting this warning when trying to compile:

```bash
In file included from /Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.cpp:12:0:
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     static void prepare (PGM_P fmt, ...);
                                        ^
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_p (PGM_P fmt, ...);
                                ^
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }
                                         ^
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.cpp:157:36: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
 void Stash::prepare (PGM_P fmt, ...) {
                                    ^
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.cpp: In static member function 'static void Stash::extract(uint16_t, uint16_t, void*)':
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.cpp:226:11: warning: 'prog_char' is deprecated (declared at /Applications/Arduino.app/Contents/Java/hardware/tools/avr/avr/include/avr/pgmspace.h:350): prog_char type is deprecated. [-Wdeprecated-declarations]
     PGM_P fmt = (PGM_P) *++segs;
           ^
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.cpp: In static member function 'static void Stash::cleanup()':
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.cpp:294:11: warning: 'prog_char' is deprecated (declared at /Applications/Arduino.app/Contents/Java/hardware/tools/avr/avr/include/avr/pgmspace.h:350): prog_char type is deprecated. [-Wdeprecated-declarations]
     PGM_P fmt = (PGM_P) *++segs;
           ^
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.cpp: At global scope:
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.cpp:318:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
 void BufferFiller::emit_p(PGM_P fmt, ...) {
                                         ^
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.cpp: In member function 'void BufferFiller::emit_p(const prog_char*, ...)':
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.cpp:367:19: warning: 'prog_char' is deprecated (declared at /Applications/Arduino.app/Contents/Java/hardware/tools/avr/avr/include/avr/pgmspace.h:350): prog_char type is deprecated. [-Wdeprecated-declarations]
             PGM_P s = va_arg(ap, PGM_P);
                   ^
In file included from /Users/xxx/Documents/Arduino/libraries/EtherCard/dhcp.cpp:17:0:
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     static void prepare (PGM_P fmt, ...);
                                        ^
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_p (PGM_P fmt, ...);
                                ^
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }
                                         ^
In file included from /Users/xxx/Documents/Arduino/libraries/EtherCard/dns.cpp:7:0:
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     static void prepare (PGM_P fmt, ...);
                                        ^
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_p (PGM_P fmt, ...);
                                ^
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }
                                         ^
In file included from /Users/xxx/Documents/Arduino/libraries/EtherCard/tcpip.cpp:14:0:
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     static void prepare (PGM_P fmt, ...);
                                        ^
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_p (PGM_P fmt, ...);
                                ^
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }
                                         ^
In file included from /Users/xxx/Documents/Arduino/libraries/EtherCard/udpserver.cpp:8:0:
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     static void prepare (PGM_P fmt, ...);
                                        ^
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_p (PGM_P fmt, ...);
                                ^
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }
                                         ^
In file included from /Users/xxx/Documents/Arduino/libraries/EtherCard/webutil.cpp:7:0:
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     static void prepare (PGM_P fmt, ...);
                                        ^
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_p (PGM_P fmt, ...);
                                ^
/Users/xxx/Documents/Arduino/libraries/EtherCard/EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }
                                         ^

Der Sketch verwendet 7.040 Bytes (22%) des Programmspeicherplatzes. Das Maximum sind 30.720 Bytes.
Globale Variablen verwenden 1.077 Bytes (52%) des dynamischen Speichers, 971 Bytes für lokale Variablen verbleiben. Das Maximum sind 2.048 Bytes.
```

I just tool the master branch and the webclient example in Arduino 1.6.8

In file included from C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.cpp:12:0:

C:\Users\Pim\Documents\Arduino\libraries\ethercard/EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     static void prepare (PGM_P fmt, ...);

                                        ^

C:\Users\Pim\Documents\Arduino\libraries\ethercard/EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     void emit_p (PGM_P fmt, ...);

                                ^

C:\Users\Pim\Documents\Arduino\libraries\ethercard/EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }

                                         ^

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.cpp:157:36: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

 void Stash::prepare (PGM_P fmt, ...) {

                                    ^

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.cpp: In static member function 'static void Stash::extract(uint16_t, uint16_t, void*)':

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.cpp:226:11: warning: 'prog_char' is deprecated (declared at c:\program files (x86)\arduino\hardware\tools\avr\avr\include\avr\pgmspace.h:350): prog_char type is deprecated. [-Wdeprecated-declarations]

     PGM_P fmt = (PGM_P) *++segs;

           ^

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.cpp: In static member function 'static void Stash::cleanup()':

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.cpp:294:11: warning: 'prog_char' is deprecated (declared at c:\program files (x86)\arduino\hardware\tools\avr\avr\include\avr\pgmspace.h:350): prog_char type is deprecated. [-Wdeprecated-declarations]

     PGM_P fmt = (PGM_P) *++segs;

           ^

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.cpp: At global scope:

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.cpp:318:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

 void BufferFiller::emit_p(PGM_P fmt, ...) {

                                         ^

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.cpp: In member function 'void BufferFiller::emit_p(const prog_char*, ...)':

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.cpp:367:19: warning: 'prog_char' is deprecated (declared at c:\program files (x86)\arduino\hardware\tools\avr\avr\include\avr\pgmspace.h:350): prog_char type is deprecated. [-Wdeprecated-declarations]

             PGM_P s = va_arg(ap, PGM_P);

                   ^

In file included from C:\Users\Pim\Documents\Arduino\libraries\ethercard\dhcp.cpp:17:0:

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     static void prepare (PGM_P fmt, ...);

                                        ^

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     void emit_p (PGM_P fmt, ...);

                                ^

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }

                                         ^

In file included from C:\Users\Pim\Documents\Arduino\libraries\ethercard\dns.cpp:7:0:

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     static void prepare (PGM_P fmt, ...);

                                        ^

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     void emit_p (PGM_P fmt, ...);

                                ^

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }

                                         ^

In file included from C:\Users\Pim\Documents\Arduino\libraries\ethercard\tcpip.cpp:14:0:

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     static void prepare (PGM_P fmt, ...);

                                        ^

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     void emit_p (PGM_P fmt, ...);

                                ^

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }

                                         ^

In file included from C:\Users\Pim\Documents\Arduino\libraries\ethercard\udpserver.cpp:8:0:

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     static void prepare (PGM_P fmt, ...);

                                        ^

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     void emit_p (PGM_P fmt, ...);

                                ^

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }

                                         ^

In file included from C:\Users\Pim\Documents\Arduino\libraries\ethercard\webutil.cpp:7:0:

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     static void prepare (PGM_P fmt, ...);

                                        ^

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     void emit_p (PGM_P fmt, ...);

                                ^

C:\Users\Pim\Documents\Arduino\libraries\ethercard\EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]

     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }

I have the latest version of library and have same issue in Arch Linux with Arduino 1.6.8.

```
In file included from /home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.cpp:12:0:
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     static void prepare (PGM_P fmt, ...);
                                        ^
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_p (PGM_P fmt, ...);
                                ^
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }
                                         ^
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.cpp:157:36: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
 void Stash::prepare (PGM_P fmt, ...) {
                                    ^
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.cpp: In static member function 'static void Stash::extract(uint16_t, uint16_t, void*)':
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.cpp:226:11: warning: 'prog_char' is deprecated (declared at /usr/share/arduino/hardware/tools/avr/avr/include/avr/pgmspace.h:350): prog_char type is deprecated. [-Wdeprecated-declarations]
     PGM_P fmt = (PGM_P) *++segs;
           ^
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.cpp: In static member function 'static void Stash::cleanup()':
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.cpp:294:11: warning: 'prog_char' is deprecated (declared at /usr/share/arduino/hardware/tools/avr/avr/include/avr/pgmspace.h:350): prog_char type is deprecated. [-Wdeprecated-declarations]
     PGM_P fmt = (PGM_P) *++segs;
           ^
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.cpp: At global scope:
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.cpp:318:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
 void BufferFiller::emit_p(PGM_P fmt, ...) {
                                         ^
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.cpp: In member function 'void BufferFiller::emit_p(const prog_char*, ...)':
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.cpp:367:19: warning: 'prog_char' is deprecated (declared at /usr/share/arduino/hardware/tools/avr/avr/include/avr/pgmspace.h:350): prog_char type is deprecated. [-Wdeprecated-declarations]
             PGM_P s = va_arg(ap, PGM_P);
                   ^
In file included from /home/ogarcia/Arduino/libraries/ethercard-master/dhcp.cpp:17:0:
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     static void prepare (PGM_P fmt, ...);
                                        ^
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_p (PGM_P fmt, ...);
                                ^
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }
                                         ^
In file included from /home/ogarcia/Arduino/libraries/ethercard-master/dns.cpp:7:0:
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     static void prepare (PGM_P fmt, ...);
                                        ^
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_p (PGM_P fmt, ...);
                                ^
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }
                                         ^
In file included from /home/ogarcia/Arduino/libraries/ethercard-master/tcpip.cpp:14:0:
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     static void prepare (PGM_P fmt, ...);
                                        ^
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_p (PGM_P fmt, ...);
                                ^
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }
                                         ^
In file included from /home/ogarcia/Arduino/libraries/ethercard-master/udpserver.cpp:8:0:
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     static void prepare (PGM_P fmt, ...);
                                        ^
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_p (PGM_P fmt, ...);
                                ^
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }
                                         ^
In file included from /home/ogarcia/Arduino/libraries/ethercard-master/webutil.cpp:7:0:
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:171:40: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     static void prepare (PGM_P fmt, ...);
                                        ^
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:240:32: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_p (PGM_P fmt, ...);
                                ^
/home/ogarcia/Arduino/libraries/ethercard-master/EtherCard.h:252:41: warning: 'prog_char' is deprecated [-Wdeprecated-declarations]
     void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }
```

--- a/EtherCard.cpp
+++ b/EtherCard.cpp
@@ -154,7 +154,7 @@ static char* wtoa (uint16_t value, char* ptr) {
 
 // write information about the fmt string and the arguments into special page/block 0    
 // block 0 is initially marked as allocated and never returned by allocateBlock 
-void Stash::prepare (PGM_P fmt, ...) {
+void Stash::prepare (const char* fmt PROGMEM, ...) {
     Stash::load(WRITEBUF, 0);
     uint16_t* segs = Stash::bufs[WRITEBUF].words;
     *segs++ = strlen_P(fmt);
@@ -187,7 +187,7 @@ void Stash::prepare (PGM_P fmt, ...) {
                 arglen = strlen((const char*) argval);
                 break;
             case 'F':
-                arglen = strlen_P((PGM_P) argval);
+                arglen = strlen_P((const char*) argval);
                 break;
             case 'E': {
                 byte* s = (byte*) argval;
@@ -223,9 +223,9 @@ void Stash::extract (uint16_t offset, uint16_t count, void* buf) {
     Stash::load(WRITEBUF, 0);
     uint16_t* segs = Stash::bufs[WRITEBUF].words;
 #ifdef __AVR__
-    PGM_P fmt = (PGM_P) *++segs;
+    const char* fmt PROGMEM = (const char*) *++segs;
 #else
-    PGM_P fmt = (PGM_P)((segs[2] << 16) | segs[1]);
+    const char* fmt PROGMEM = (const char*)((segs[2] << 16) | segs[1]);
     segs += 2;
 #endif
     Stash stash;
@@ -291,9 +291,9 @@ void Stash::cleanup () {
     Stash::load(WRITEBUF, 0);
     uint16_t* segs = Stash::bufs[WRITEBUF].words;
 #ifdef __AVR__
-    PGM_P fmt = (PGM_P) *++segs;
+    const char* fmt PROGMEM = (const char*) *++segs;
 #else
-    PGM_P fmt = (PGM_P)((segs[2] << 16) | segs[1]);
+    const char* fmt PROGMEM = (const char*)((segs[2] << 16) | segs[1]);
     segs += 2;
 #endif
     for (;;) {
@@ -315,7 +315,7 @@ void Stash::cleanup () {
     }
 }
 
-void BufferFiller::emit_p(PGM_P fmt, ...) {
+void BufferFiller::emit_p(const char* fmt PROGMEM, ...) {
     va_list ap;
     va_start(ap, fmt);
     for (;;) {
@@ -364,7 +364,7 @@ void BufferFiller::emit_p(PGM_P fmt, ...) {
             strcpy((char*) ptr, va_arg(ap, const char*));
             break;
         case 'F': {
-            PGM_P s = va_arg(ap, PGM_P);
+            const char* s PROGMEM = va_arg(ap, const char*);
             char d;
             while ((d = pgm_read_byte(s++)) != 0)
                 *ptr++ = d;

--- a/EtherCard.h
+++ b/EtherCard.h
@@ -168,7 +168,7 @@ public:
     //   offs = 63;
     // }
 
-    static void prepare (PGM_P fmt, ...);
+    static void prepare (const char* fmt PROGMEM, ...);
     static uint16_t length ();
     static void extract (uint16_t offset, uint16_t count, void* buf);
     static void cleanup ();
@@ -181,7 +181,7 @@ public:
 *
 *   This class provides formatted printing into memory. Users can use it to write into send buffers.
 *
-*   Nota: PGM_P: is a pointer to a string in program space (defined in the source code)
+*   Nota: PGM_P: is a pointer to a string in program space (defined in the source code, updated to PROGMEM)
 *
 *   # Format string
 *
@@ -237,7 +237,7 @@ public:
     *   @param  fmt Format string (see Class description)
     *   @param  ... parameters for format string
     */
-    void emit_p (PGM_P fmt, ...);
+    void emit_p (const char* fmt PROGMEM, ...);
 
     /** @brief  Add data to buffer from main memory
     *   @param  s Pointer to data
@@ -249,7 +249,7 @@ public:
     *   @param  p Program space string pointer
     *   @param  n Number of characters to copy
     */
-    void emit_raw_p (PGM_P p, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }
+    void emit_raw_p (const char* p PROGMEM, uint16_t n) { memcpy_P(ptr, p, n); ptr += n; }
 
     /** @brief  Get pointer to start of buffer
     *   @return <i>uint8_t*</i> Pointer to start of buffer

--- a/examples/noipClient/noipClient.ino
+++ b/examples/noipClient/noipClient.ino
@@ -274,6 +274,6 @@ void checkNoIPResponse() {
     }
 }
 
-void SerialPrint_P(PGM_P str) {
+void SerialPrint_P(const char* str PROGMEM) {
     for (uint8_t c; (c = pgm_read_byte(str)); str++) Serial.write(c);
 }

